src/aura/Timesheets/TimesheetsHelper.js

```diff
@@ -1,13 +1,40 @@
 ({
-	getLastTimesheets : function(component,timesheets) {
-        var id="";        
-        var lastT=component.get("v.lasttimesheets");
-        for(var ts of timesheets ){
-            if(ts.Employee__r.Id!=id){
-               id= ts.Employee__r.Id;
-               lastT.push(ts);
-            }
-        }        
-        component.set("v.lasttimesheets",lastT);
+	
+    getAllTimesheets : function(component) {
+        // Create the action
+       var action = component.get("c.getTimesheets");
+       
+	   // Add callback behavior for when response is received
+	   action.setCallback(this, function(response) {
+
+		   var state = response.getState();
+		   if (state === "SUCCESS") {
+               
+               var mode = component.get("v.listmode"); 
+               var timesheets=response.getReturnValue();
+               
+               if(mode=="all"){                    
```

Comment:
you should actually change the timesheets loaded from the server based on the mode, not filter out  results based on the mode.